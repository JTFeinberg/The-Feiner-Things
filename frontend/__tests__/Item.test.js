import React from 'react'
import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This is a really cool item!',
  image: 'item.jpg',
  largeImage: 'largeItem.jpg'
}

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const price = '$50.35'
    expect(price).toMatchSnapshot()
  })
  // const wrapper = shallow(<ItemComponent item={fakeItem} />)

  // it('renders the price tage and title properly', () => {
  //   const PriceTag = wrapper.find('PriceTag')
  //   expect(PriceTag.children().text()).toBe('$50')
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
  // })

  // it('renders the image properly', () => {
  //   const img = wrapper.find('img')
  //   expect(img.props().src).toBe(fakeItem.image)
  //   expect(img.props().alt).toBe(fakeItem.title)
  // })

  // it('renders out the buttons properly', () => {
  //   const buttonList = wrapper.find('.buttonList')
  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link')).toHaveLength(1)
  //   expect(buttonList.find('AddToCart').exists()).toBe(true)
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true)
  // })
})
