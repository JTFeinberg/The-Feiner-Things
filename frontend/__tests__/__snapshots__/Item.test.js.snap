// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    item={
      Object {
        "description": "This is a really cool item!",
        "id": "ABC123",
        "image": "item.jpg",
        "largeImage": "largeItem.jpg",
        "price": 5000,
        "title": "A Cool Item",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="A Cool Item"
          src="item.jpg"
        />,
        <Title>
          <Link
            href={
              Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC123",
                },
              }
            }
          >
            <a>
              A Cool Item
            </a>
          </Link>
        </Title>,
        <PriceTag>
          $50
        </PriceTag>,
        <p>
          This is a really cool item!
        </p>,
        <div
          className="buttonList"
        >
          <Link
            href={
              Object {
                "pathname": "/update",
                "query": Object {
                  "id": "ABC123",
                },
              }
            }
          >
            <a>
              Edit ✏️
            </a>
          </Link>
          <AddToCart
            id="ABC123"
          />
          <DeleteItem
            id="ABC123"
          >
            Delete This Item
          </DeleteItem>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "A Cool Item",
          "src": "item.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            href={
              Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC123",
                },
              }
            }
          >
            <a>
              A Cool Item
            </a>
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a>
              A Cool Item
            </a>,
            "href": Object {
              "pathname": "/item",
              "query": Object {
                "id": "ABC123",
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A Cool Item",
            },
            "ref": null,
            "rendered": "A Cool Item",
            "type": "a",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "$50",
        },
        "ref": null,
        "rendered": "$50",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a really cool item!",
        },
        "ref": null,
        "rendered": "This is a really cool item!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href={
                Object {
                  "pathname": "/update",
                  "query": Object {
                    "id": "ABC123",
                  },
                }
              }
            >
              <a>
                Edit ✏️
              </a>
            </Link>,
            <AddToCart
              id="ABC123"
            />,
            <DeleteItem
              id="ABC123"
            >
              Delete This Item
            </DeleteItem>,
          ],
          "className": "buttonList",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                Edit ✏️
              </a>,
              "href": Object {
                "pathname": "/update",
                "query": Object {
                  "id": "ABC123",
                },
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit ✏️",
              },
              "ref": null,
              "rendered": "Edit ✏️",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "ABC123",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete This Item",
              "id": "ABC123",
            },
            "ref": null,
            "rendered": "Delete This Item",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="A Cool Item"
            src="item.jpg"
          />,
          <Title>
            <Link
              href={
                Object {
                  "pathname": "/item",
                  "query": Object {
                    "id": "ABC123",
                  },
                }
              }
            >
              <a>
                A Cool Item
              </a>
            </Link>
          </Title>,
          <PriceTag>
            $50
          </PriceTag>,
          <p>
            This is a really cool item!
          </p>,
          <div
            className="buttonList"
          >
            <Link
              href={
                Object {
                  "pathname": "/update",
                  "query": Object {
                    "id": "ABC123",
                  },
                }
              }
            >
              <a>
                Edit ✏️
              </a>
            </Link>
            <AddToCart
              id="ABC123"
            />
            <DeleteItem
              id="ABC123"
            >
              Delete This Item
            </DeleteItem>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "A Cool Item",
            "src": "item.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              href={
                Object {
                  "pathname": "/item",
                  "query": Object {
                    "id": "ABC123",
                  },
                }
              }
            >
              <a>
                A Cool Item
              </a>
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                A Cool Item
              </a>,
              "href": Object {
                "pathname": "/item",
                "query": Object {
                  "id": "ABC123",
                },
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A Cool Item",
              },
              "ref": null,
              "rendered": "A Cool Item",
              "type": "a",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "$50",
          },
          "ref": null,
          "rendered": "$50",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a really cool item!",
          },
          "ref": null,
          "rendered": "This is a really cool item!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href={
                  Object {
                    "pathname": "/update",
                    "query": Object {
                      "id": "ABC123",
                    },
                  }
                }
              >
                <a>
                  Edit ✏️
                </a>
              </Link>,
              <AddToCart
                id="ABC123"
              />,
              <DeleteItem
                id="ABC123"
              >
                Delete This Item
              </DeleteItem>,
            ],
            "className": "buttonList",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a>
                  Edit ✏️
                </a>,
                "href": Object {
                  "pathname": "/update",
                  "query": Object {
                    "id": "ABC123",
                  },
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit ✏️",
                },
                "ref": null,
                "rendered": "Edit ✏️",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "ABC123",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete This Item",
                "id": "ABC123",
              },
              "ref": null,
              "rendered": "Delete This Item",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
